presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -157,11 +157,26 @@ public SemiTransactionalHiveMetastore(
     }
 
     public synchronized Optional<Table> getTable(String databaseName, String tableName)
+    {
+        return getTable(databaseName, tableName, Optional.empty());
+    }
+
+    public synchronized Optional<Table> getTable(String databaseName, String tableName, Optional<ConnectorSession> session)
     {
         checkReadable();
         Action<TableAndMore> tableAction = tableActions.get(new SchemaTableName(databaseName, tableName));
         if (tableAction == null) {
-            return delegate.getTable(databaseName, tableName);
+            // TODO do it better – probably add semantic type instead of remove in case of leaks
+            Optional<Table> table = delegate.getTable(databaseName, tableName);
+            // TODO cannot import HiveSessionProperties class here to do this check because of circular dependency on presto-hive
```

Comment:
We've never used `session` before in any metastore related code. Adding this dependency is generally not desirable.